### PR TITLE
docs(tutorial): update PhoneCat tutorial to Angular 1.5, components and style guide

### LINKED ISSUE #13312: Phonecat tutorial does not follow style guide
The [Phonecat tutorial](https://docs.angularjs.org/tutorial) does not really follow the conventions laid out by the [Angular Styleguide](https://github.com/johnpapa/angular-styleguide).

In particular, I think the following rules should be applied:

* [Rule of 1](https://github.com/johnpapa/angular-styleguide#style-y001)
* [Folders-by-feature structure](https://github.com/johnpapa/angular-styleguide#style-y152)
* [controllerAs View Syntax](https://github.com/johnpapa/angular-styleguide#style-y030)

It also might not hurt to explicitly refer to the style guide in the tutorial.

### PR DESCRIPTION
**Pull request by teropa**
Bring the PhoneCat tutorial up to speed with Angular 1.5 and the modern
component style. Introduce components instead of bare ngControllers from
the get-go. Add an additional chapter about organizing the code in feature modules
and using the style guide.

This also makes the guide compatible with the Angular 2 upgrade tutorial
on angular.io.

Closes #13312
Related to angular/angular-phonecat#289

### CODE REVIEW

**Comment by teropa** [comment]
@petebacondarwin this together with the rebased steps in angular/angular-phonecat#289 completes the update from my point of view, except for the diagram images. Hoping the source files for those are somewhere, but it should be also possible to redo them from scratch.

**Comment by petebacondarwin** [comment]
@teropa - thanks for this. We will take a look this week and hopefully update the tutorial!

**Comment by teropa** [comment]
Cool, let me know if you need anything.

File: docs/content/tutorial/step_02.ngdoc [hunk 1]
```diff
@@ -230,25 +232,25 @@ browser is limited, which results in your karma tests running extremely slow.
 
 # Experiments
 
-* Add another binding to `index.html`. For example:
+* Add another binding to `phone-list.html`. For example:
 
   ```html
-  <p>Total number of phones: {{phones.length}}</p>
+  <p>Total number of phones: {{ctrl.phones.length}}</p>
```
Comment (dmitriz) [inline docs/content/tutorial/step_02.ngdoc]: You meant `$ctrl`?

**Comment by Narretz** [comment]
This is closed by https://github.com/angular/angular.js/commit/b5dbc9834a18d35470ff3ba8d3bcc54866183b41 Thanks again both @teropa and @gkalpak

**Pull request closed without merging**